cli/src/cmd/forge/script/broadcast.rs

```diff
@@ -0,0 +1,260 @@
+        local_wallets =
+            local_wallets.into_iter().map(|wallet| wallet.with_chain_id(chain)).collect();
+
+        // in case of --force-resume, we forgive the first nonce disparity of each from
+        let mut nonce_offset: BTreeMap<Address, U256> = BTreeMap::new();
+
+        // Iterate through transactions, matching the `from` field with the associated
+        // wallet. Then send the transaction. Panics if we find a unknown `from`
+        let sequence = deployment_sequence.clone();
+        let sequence =
+            sequence.transactions.range((deployment_sequence.index as usize)..).map(|tx| {
+                let from = into_legacy_ref(tx)?.from.expect("No sender for onchain transaction!");
+                if let Some(wallet) =
+                    local_wallets.iter().find(|wallet| (**wallet).address() == from)
+                {
+                    let signer = SignerMiddleware::new(provider.clone(), wallet.clone());
+                    Ok((tx.clone(), signer))
+                } else {
+                    Err(eyre::eyre!(format!(
+                        "No associated wallet for address: {:?}. Unlocked wallets: {:?}",
+                        from,
+                        local_wallets
+                            .iter()
+                            .map(|wallet| wallet.address())
+                            .collect::<Vec<Address>>()
+                    )))
+                }
+            });
+
+        for payload in sequence {
+            match payload {
+                Ok((tx, signer)) => {
+                    let mut legacy_or_1559 =
+                        if is_legacy { tx } else { TypedTransaction::Eip1559(into_1559(tx)?) };
+                    set_chain_id(&mut legacy_or_1559, chain)?;
+
+                    let from = *legacy_or_1559.from().expect("no sender");
+                    match foundry_utils::next_nonce(from, &fork_url, None) {
+                        Ok(nonce) => {
+                            let tx_nonce = *legacy_or_1559.nonce().expect("no nonce");
+                            let offset = if self.force_resume {
+                                match nonce_offset.get(&from) {
+                                    Some(offset) => *offset,
+                                    None => {
+                                        let offset = nonce - tx_nonce;
+                                        nonce_offset.insert(from, offset);
+                                        offset
+                                    }
+                                }
+                            } else {
+                                U256::from(0u32)
+                            };
+
+                            if nonce != tx_nonce + offset {
+                                deployment_sequence
+                                    .save()
+                                    .expect("not able to save deployment sequence");
```

Comment:
Not sure if this works, but maybe you can `impl Drop for ScriptSequence` and call `self.save().expect(..)` in it? Might saves you from having to manually save every time before failing